src/overlays/actors/ovl_En_Talk_Gibud/z_en_talk_gibud.c

```diff
@@ -28,166 +83,1065 @@ const ActorInit En_Talk_Gibud_InitVars = {
+    }
+    if (EnTalkGibud_PlayerInRangeWithCorrectState(this, globalCtx)) {
+        if (gSaveContext.playerForm != PLAYER_FORM_GORON && gSaveContext.playerForm != PLAYER_FORM_DEKU &&
+            Actor_IsActorFacingLink(&this->actor, 0x38E3)) {
+            EnTalkGibud_SetupWalkToPlayer(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFEFD4.s")
+void EnTalkGibud_SetupStunned(EnTalkGibud* this) {
+    this->stunTimer = 10;
+    this->actor.speedXZ = 0.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    if (this->effectTimer != 0) {
+        func_800BCB70(&this->actor, 0, 0xC8, 0, 0x28);
+    } else {
+        func_800BCB70(&this->actor, 0, 0xC8, 0, 0x28);
+    }
+    this->actionFunc = EnTalkGibud_Stunned;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF030.s")
+void EnTalkGibud_Stunned(EnTalkGibud* this, GlobalContext* globalCtx) {
+    if (this->actor.colorFilterTimer == 0) {
+        if (this->actor.colChkInfo.health == 0) {
+            EnTalkGibud_SetupDead(this);
+        } else {
+            EnTalkGibud_SetupDamage(this);
+        }
+    }
+    if (this->stunTimer != 0) {
+        this->stunTimer--;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF22C.s")
+void EnTalkGibud_SetupDamage(EnTalkGibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, EN_TALK_GIBUD_ANIMATION_DAMAGE);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_REDEAD_DAMAGE);
+    this->stunTimer = 0;
+    this->grabWaitTimer = 0;
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    this->actionFunc = EnTalkGibud_Damage;
+    this->actor.speedXZ = -2.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF288.s")
+void EnTalkGibud_Damage(EnTalkGibud* this, GlobalContext* globalCtx) {
+    if (this->actor.speedXZ < 0.0f) {
+        this->actor.speedXZ += 0.15f;
+    }
+    if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
+        this->unk_3F7 = -1;
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        if (this->effectTimer > 0 && this->effectType == 0 && this->type == EN_TALK_GIBUD_TYPE_GIBDO) {
+            this->actor.hintId = 0x2A;
+            this->actor.flags &= ~(0x8 | 0x1);
+            this->actor.flags |= (0x4 | 0x1);
+            SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06010B88, NULL, this->jointTable, this->morphTable, 26);
+            this->type = EN_TALK_GIBUD_TYPE_REDEAD;
+        }
+        if (EnTalkGibud_PlayerOutOfRange(this, globalCtx)) {
+            EnTalkGibud_SetupWalkToHome(this);
+        } else {
+            EnTalkGibud_SetupWalkToPlayer(this);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Talk_Gibud/func_80AFF330.s")
+void EnTalkGibud_SetupDead(EnTalkGibud* this) {
+    func_800BDC5C(&this->skelAnime, sAnimations, EN_TALK_GIBUD_ANIMATION_DEATH);
+    this->actor.flags &= -2;
```

Comment:
```suggestion
    this->actor.flags &= ~1;
```

Reply:
Oops, I didn't notice this because I read it as `~2`. Thanks